animate/metric.py

```diff
@@ -128,19 +133,58 @@ def _process_parameters(self, metric_parameters):
         return mp, vp
 
     def set_parameters(self, metric_parameters={}):
-        """
-        Set metric parameter values internally.
-
-        :kwarg metric_parameters: a dictionary of parameters to be passed to PETSc's
-            Riemannian metric implementation. All such options have the prefix
-            `dm_plex_metric_`.
+        r"""
+        Set metric parameter values internally. All options have the prefix
+        `dm_plex_metric_` and are listed below (with prefix dropped for brevity):
+
+        * `h_min`: Minimum tolerated metric magnitude, which allows approximate control
+            of minimum element size in the adapted mesh. Supports :class:`~.Constant`
+            and :class:`~.Function` input, as well as :class:`float`. Default: 1.0e-30.
+        * `h_max`: Maximum tolerated metric magnitude, which allows approximate control
+            of maximum element size in the adapted mesh. Supports :class:`~.Constant`
+            and :class:`~.Function` input, as well as :class:`float`. Default: 1.0e+30.
+        * `a_max`: Maximum tolerated metric anisotropy, which allows approximate control
+            of maximum element anisotropy in the adapted mesh. Supports
+            :class:`~.Constant` and :class:`~.Function` input, as well as :class:`float`.
+            Default: 1.0e+05.
+        * `p`: :math:`L^p` normalisation order. Supports ``np.inf`` as well as
+            :class:`float` values from :math:`[0,\infty)`. Default: 1.0.
+        * `target_complexity`: Strictly positive target metric complexity value. No
+            default - **must be set**.
+        * `gradation_factor`: Maximum ratio by which adjacent edges in the adapted mesh
+            may differ. Default: 1.3.
+        * `hausdorff_number`: Spatial scale factor for the problem. Default: 0.01.
+        * `boundary_tag`: Mesh boundary tag to restrict attention to during
+            boundary-specific manipulations. Unset by default, which implies all
+            boundaries are considered. (Note that this parameter does not currently exist
```

Comment:
It might be good to add how this parameter currently interacts with a similar flag which can be set in a mesh originating from a `.gmsh`, as Firedrake does seem to preserve that flag when communicating between PETSc and MMG. Is this a different flag altogether?